salt/transport/base.py

```diff
@@ -374,9 +376,23 @@ class DaemonizedPublishServer(PublishServer):
     PublishServer that has a daemon associated with it.
     """
 
-    def pre_fork(self, process_manager):
+    @abstractmethod
+    def __init__(
+        self,
+        opts,
+        pub_host=None,
+        pub_port=None,
+        pub_path=None,
+        pull_host=None,
+        pull_port=None,
+        pull_path=None,
+        pull_path_perms=0o600,
+        pub_path_perms=0o600,
+        ssl=None,
```

Comment:
ssl is not a valid option for all transports.

Reply:
Then it should be ignored by that transport. We already have [issues](https://github.com/saltstack/salt/issues/66577#issuecomment-2251245303) with trying to pass the 'ssl' config when a transport does not support it. Do you expect the [factory methods](https://github.com/saltstack/salt/blob/ca322bdd1d6cfc1f8d935120e387b6d3b8e09a83/salt/transport/base.py#L99-L119) to know which transports do and do not support SSL and not pass it along?

Reply:
I'd argue that it's the whole point of using a factory method rather than just instantiating the class directly.